pkg/kvstore/etcd.go

```diff
@@ -59,7 +59,7 @@ var (
 
 	// initialConnectionTimeout  is the timeout for the initial connection to
 	// the etcd server
-	initialConnectionTimeout = 3 * time.Minute
+	initialConnectionTimeout = 15 * time.Minute
```

Comment:
This is fine for now, maybe make this configurable as a kvstore option in the future?

Reply:
Good idea. I'll file a GH issue